### PR TITLE
Popular artist container

### PR DESCRIPTION
**Pull request by yuki24**
This (partially) implements https://github.com/artsy/collector-experience/issues/505

Right now typing in the search bar doesn't update the list. Also, a click event for each individual artist is not hooked up. But as we discussed we should get this merged before we hook up the data layer with all the components.

screencapture-localhost-9001-1507448346326

 * [ ] Use the right icon for (+)
 * [x] after clicking an artist row should fade out
 * [ ] a new similar artist should fade in after following an artist
 * [ ] Actually use [the related artist endpoint](https://github.com/artsy/gravity/blob/d3eaf52632221d0a9cc95877accb8082375e26f8/app/api/v1/related_endpoint.rb#L7) to show a similar artist
 * [ ] Search and show new results as you type
 * [ ] Show "No results found" when there's no result
 * [ ] Clicking the `x` should clear out the search bar
 * [ ] The search result view should be scrollable when there are many results
 * [ ] Tweak the paddings/margins to match what's specified in the mockup
 * [ ] Change the title/subtitle size/margin/padding on mobile (could be done separately)
 * [ ] The "next" button should stay at the bottom on mobile (could be done separately)

### CODE REVIEW

File: src/components/onboarding/__stories__/artist_selector.story.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { storiesOf } from "@storybook/react"
+import * as React from "react"
+
+import Artists from "../steps/artists";
+
+storiesOf("Onboarding").add("Artist Selector", () => <Artists onNextButtonPressed={null} />)
```
Comment (yuki24) [inline src/components/onboarding/__stories__/artist_selector.story.tsx]: I wasn't sure if it's right to do `onNextButtonPressed={null}`. Obviously, we don't any callbacks in the storybook, but passing in an empty function seems to make `yarn lint` angry.

Comment (alloy) [reply src/components/onboarding/__stories__/artist_selector.story.tsx]: You can do e.g. `() => null`

File: src/components/onboarding/popular_artist_list.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        }
+      }
+    `,
+  },
+})
+
+export default function() {
+  Relay.injectNetworkLayer(artsyNetworkLayer())
+  return <Relay.RootContainer Component={PopularArtistsContainer} route={new PopularArtistQueryConfig()} />
+}
```
Comment (alloy) [inline src/components/onboarding/popular_artist_list.tsx]: This shouldn’t be in this file, but instead be used in the places where you want to mount the tree, in this case in the story file.

File: src/components/onboarding/popular_artist_list.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+const PopularArtistsContainer = Relay.createContainer(PopularArtists, {
+  fragments: {
+    popular_artists: () => Relay.QL`
+      fragment on PopularArtists {
+        artists {
+          name
+          image {
+            cropped(width: 50, height: 50) {
+              url
+            }
```
[the remark this answers is not in the document]
Comment (alloy) [reply src/components/onboarding/popular_artist_list.tsx]: Yup, this looks fine. My only question is if this image is gong to be displayed at 50x50? If so, then for retina displays it should fetch 100x100 images instead.

File: src/components/onboarding/popular_artist_list.tsx (hunk 2 above)
Comment (alloy) [reply src/components/onboarding/popular_artist_list.tsx]: Export the result of `Relay.createContainer` instead.

File: src/components/onboarding/popular_artist_list.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuki24) [reply src/components/onboarding/popular_artist_list.tsx]: Should we change the size depending on the display density, or is it ok to always use 100x100?

[the remark this answers is not in the document]
Comment (alloy) [reply src/components/onboarding/popular_artist_list.tsx]: iirc Force has always just fetched the larger version /cc @craigspaeth

[the remark this answers is not in the document]
Comment (craigspaeth) [reply src/components/onboarding/popular_artist_list.tsx]: Yeah, we've typically just used the larger size especially pre-on-the-fly resizing when we didn't really have a choice.

[the remark this answers is not in the document]
Comment (yuki24) [reply src/components/onboarding/popular_artist_list.tsx]: 👍 I'll change that to 100x100.

In the future, we should probably implement a component that picks up the appropriate size of image for the screen's density for further optimization:

```jsx
<Image src="http:s//..." 2x="http:s//..." 3x="http:s//...">
```

**Comment by alloy** [comment]
Merging this now as-is so that I can rename the files without causing you a headache.

File: src/components/onboarding/popular_artist_list.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (alloy) [reply src/components/onboarding/popular_artist_list.tsx]: Aye. Also look at the inage component in Emission which takes both screen pixel ratio and the laid out size of the image component to fetch exactly the right size.

**Pull request merged**